src/param.rs

```diff
@@ -11,6 +11,49 @@ use crate::{AtomicF32, SampleRate, RENDER_QUANTUM_SIZE};
 
 use crossbeam_channel::{Receiver, Sender};
 
+// arguments sanity check functions for automation methods
+fn assert_non_negative(value: f64, method_name: &str, arg_name: &str) {
```

Comment:
Maybe add a `#[track_caller]` attribute?

Reply:
I will send a PR soon to enable that attribute on our other assertions, and to assert channel counts in more places